addons/website_sale/views/templates.xml

```diff
@@ -486,14 +485,14 @@
                                 <input type="hidden" name="csrf_token" t-att-value="request.csrf_token()" />
                                 <div class="js_product js_main_product">
                                     <t t-placeholder="select">
-                                        <input type="hidden" class="product_id" name="product_id" t-att-value="product.product_variant_id.id if len(product.product_variant_ids) == 1 else '0'" />
+                                        <input type="hidden" class="product_id" name="product_id" t-att-value="product.filtered_product_variant_id.id if len(product.filtered_product_variant_ids) == 1 else '0'" />
                                         <input type="hidden" class="product_template_id" name="product_template_id" t-att-value="product.id" />
                                         <t t-call="sale.variants">
                                             <t t-set="ul_class" t-value="'flex-column'" />
                                         </t>
                                     </t>
                                     <t t-call="website_sale.product_price" />
-                                    <p t-if="product.has_dynamic_attributes() or len(product.product_variant_ids) &gt; 1" class="css_not_available_msg bg-danger" style="padding: 15px;">This combination does not exist.</p>
+                                    <p t-if="product.has_dynamic_attributes() or len(product.filtered_product_variant_ids) &gt; 1" class="css_not_available_msg alert alert-warning">This combination does not exist.</p>
```

Comment:
@seb-odoo 

I don't think you should use "filtered_product_variant_ids" here because if there is only one combination available we still need to display the "This combination does not exist." message.

(This should not happen in "regular" configurations but better safe than sorry).

Reply:
I originally just replaced them all on website_ without paying attention.

I need to rework filtered_product_variant_ids because I can't add a field on stable. I'll create a method calculating it, and review all the places where it is used at that point.

I'll pay special attention to this one then.

Reply:
I reworked this, I'm using first combination everywhere instead of filtered_product_variant_ids now. Just the list view of variants has to use the filtered_product_variant_ids.